File: Week 01/id_454/LeetCode_189_454.php
<?php class Solution
{
    /**
     * 三次翻转
     *
     * @param [type] $nums
     * @param [type] $k
     * @return void
     */
    function rotate(&$nums, $k)
    {
        $k = $k % count($nums);
        $this->reverse($nums, 0, count($nums) - 1);
        $this->reverse($nums, 0, $k - 1);
        $this->reverse($nums, $k, count($nums) - 1);
    }

    function reverse(&$nums, $start, $end)
    {
        while ($start < $end) {
            $tmp = $nums[$start];
            $nums[$start] = $nums[$end];
            $nums[$end] = $tmp;
            $start++;
            $end--;
        }
    }
}

File: Week 01/id_454/LeetCode_189_454.go
package main

//三次翻转
func rotate(nums []int, k int)  {
    k = k%len(nums)
    reverse(nums[:len(nums)-k])
    reverse(nums[len(nums)-k:])
    reverse(nums)
}

func reverse(nums []int) {  
    i,j := 0, len(nums)-1
    for i<j {
        nums[i],nums[j] = nums[j],nums[i]
        i++
        j--
    }
}

